tools/codegen/api/python.py

```diff
@@ -602,6 +602,10 @@ def argument_type_str(t: Type, *, simple_type: bool = False) -> str:
         if str(t.elem) == 'Tensor':
             # Is it desired to keep '?' for simple_type with new style dispatcher?
             return 'Tensor?'
+        # int[]? resolves to a specialized OptionalIntArrayRef class
+        if hasattr(t.elem, 'elem') and str(t.elem.elem) == 'int':
+            size = t.elem.size if not simple_type else None
+            return f'OptionalIntArrayRef[{size}]' if size is not None else 'OptionalIntArrayRef'
```

Comment:
I think `OptionalArrayRef<T>` makes sense for all types, not just `int`. e.g. a `float[]?` called with an `int` would likely have the exact same dangling pointer issue.

Reply:
That's true. Alright, I'll focus on fixing the remaining issues with `int[]? --> OptionalIntArrayRef`, and after that's fully working, I'll generalize for all `OptionalArrayRef<T>`